File: CodeSolutions/167. Two Sum II - Input array is sorted/167.java
# Language: Python
# Commit Timestamp: 2022-07-28T00:00:00-03:00
# Solution Logic: Placeholder for explanation.

# Language: Python
# Commit Timestamp: 2022-07-28T00:00:00-03:00
# Solution Logic: Placeholder for explanation.

# Language: Python
# Commit Timestamp: 2022-07-28T00:00:00-03:00
# Solution Logic: Placeholder for explanation.

class Solution {
  public int[] twoSum(int[] numbers, int target) {
    int l = 0;
    int r = numbers.length - 1;

    while (numbers[l] + numbers[r] != target)
      if (numbers[l] + numbers[r] < target)
        ++l;
      else
        --r;

    return new int[] {l + 1, r + 1};
  }
}

File: CodeSolutions/167. Two Sum II - Input array is sorted/167.py
# Language: Python
# Commit Timestamp: 2022-08-27T00:00:00-03:00
# Solution Logic: Placeholder for explanation.

# Language: Python
# Commit Timestamp: 2022-08-27T00:00:00-03:00
# Solution Logic: Placeholder for explanation.

# Language: Python
# Commit Timestamp: 2022-08-27T00:00:00-03:00
# Solution Logic: Placeholder for explanation.

class Solution:
  def twoSum(self, numbers: list[int], target: int) -> list[int]:
    l = 0
    r = len(numbers) - 1

    while l < r:
      summ = numbers[l] + numbers[r]
      if summ == target:
        return [l + 1, r + 1]
      if summ < target:
        l += 1
      else:
        r -= 1
